File: Medium/2390_Removing_Stars_From_a_String/Solution.java
import java.util.ArrayDeque;
import java.util.Deque;

public class Solution {
    public String removeStars(String s) {
        Deque<Character> st = new ArrayDeque<>();

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '*' && st.size() != 0) {
                st.pop();
            } else if (s.charAt(i) != '*') {
                st.push(s.charAt(i));
            }
        }
        
        StringBuilder sb = new StringBuilder();
        while (!st.isEmpty()) {
            sb.append(st.pop());
        }
        return sb.reverse().toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        String s1 = "leet**cod*e";
        System.out.println(solution.removeStars(s1));

        // test cases 2
        String s2 = "erase*****";
        System.out.println(solution.removeStars(s2));
        
    }
}

File: Medium/2390_Removing_Stars_From_a_String/Solution.go
package main

import "fmt"

// Past the function from leetcode here
func removeStars(s string) string {
	st := []rune{}

	for _, c := range s {
		if c == '*' && len(st) > 0 {
			st = st[:len(st)-1]
		} else if c != '*' {
			st = append(st, c)
		}
	}

	return string(st)
}

func main() {
	// test cases 1
	s1 := "leet**cod*e"
	fmt.Println(removeStars(s1))

	// test cases 2
	s2 := "erase*****"
	fmt.Println(removeStars(s2))
}
